### DISCUSSION TITLE
Overriding master realm configuration

### DISCUSSION THREAD

**Discussion by SmithJosh** [keycloak/keycloak#16087] [Q&A]
I'm deploying Keycloak to K8s using the operator. I would like to be able to override certain master realm configurations and have those be applied automatically so I don't have to manually configure Keycloak in each cluster where I install it. I'm struggling to find a way to do that. Here's what I've tried 

* Keycloak operator: The KeycloakRealmImport works only for new realms. It won't override the master realm config.
* Import realm on startup: Keycloak also refuses to override the master realm with the --import-realm flag ( https://www.keycloak.org/server/importExport#_importing_a_realm_during_startup).
* Import realm in Dockerfile for custom keycloak image: As far as I can tell, there realm imports have to happen at runtime, or at least the import feature has to be able to connect to whatever database Keycloak is using. So importing in a Dockerfile isn't an option at build time.

I also saw [this discussion](https://github.com/keycloak/keycloak/discussions/11027) and tried to import a custom realm then set KC_SPI_ADMIN_REALM to use that new realm, but no luck there either. Keycloak creates a realm with the specified name and default master realm configuration, and then rejects the import.

What's the recommended way to customize master realm configurations in production?

**Comment by jaceblake** [comment]
You can do it with ansible https://docs.ansible.com/ansible/latest/collections/community/general/keycloak_client_module.html
